### ISSUE TITLE
[Feature Request] Ability to run an Stream inside an isolate and communicate with a callback

### ISSUE THREAD

**Issue by JCKodel** [lohnn/integral_isolates#13]
My use case: I have an Isar database running on isolate (this isolate never ends, it starts with the app).

Problem is: Isar have a query watcher that results in a Stream of some type. But I can't return Streams from isolates nor pass callbacks to it, because they are in separate memory space.

So, it would be nice if this could be implemented:

```dart
abstract class Repository {
  const Repository();

  static final _isolated = Isolated(backpressureStrategy: NoBackPressureStrategy(), autoInit: true);

  static Stream<int> _staticStreamCallback(String someParameter) async* {
    yield someParameter.length;
  }

  Future<Stream<String>> listenToIsolate(String someParameter) async {
    return _isolated.isolateStream<int>(_staticStreamCallback, someParameter);
  }
}
```

`isolateStream<T>` would create a `StreamController<T>`, then run the user code stream (my _staticStreamCallback), piping each `yield` to the outside world.

This would allow us to run a Stream method in an isolate, where each `yield` would send the response to the SendPort, thus, reaching the main thread stream.

**Comment by lohnn** [comment]
So this is a compliment to the `isolate` function, but except this returns multiple values through a stream, and once the stream is finished you continue as uaual?

I believe that you wouldn't have to actually return a `Future<Stream<T>>`, where a `Stream<T>` would suffice as it is inherently already asynchronous.

I really like the idea and I'll have a think about it and see how to most properly implement this! 👍

**Comment by lohnn** [comment]
I have slowly started working on this, you can follow the progress in the branch feat/isolate_stream :)
Haven't had heaps of time looking into this, and I am also trying to finish off my tailored isolate work. But please have a look and write some feedback if you got some 👍

**Comment by lohnn** [comment]
Hey @JCKodel, I have now implemented a version with streams now and I'll try to release a new version to pub.dev later today.
I would be very happy if you could test it and see if it works as intended after that.

And thank you for the great idea! Loved it 🎉

**Comment by lohnn** [comment]
I'm closing this now, as there is an implementation for this now. Hopefully it works the way you want it too, and if it does not, please don't hesitate open another issue 👍
